examples/bri-3/src/bri/identity/bpiAccounts/agents/bpiAccounts.agent.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { v4 } from 'uuid';
+
+import { NOT_FOUND_ERR_MESSAGE } from '../api/err.messages';
+import { BpiAccount } from '../models/bpiAccount';
+
+import { BpiAccountStorageAgent } from './bpiAccountsStorage.agent';
+
+// Agent methods have extremely declarative names and perform a single task
+@Injectable()
+export class BpiAccountAgent {
+  constructor(private repo: BpiAccountStorageAgent) {}
+  public throwIfCreateBpiAccountInputInvalid() {
+    return true;
+  }
+
+  public createNewBpiAccount(nonce: string): BpiAccount {
+    return new BpiAccount(v4(), nonce, []);
+  }
+
+  public async fetchUpdateCandidateAndThrowIfUpdateValidationFails(
```

Comment:
I'm actually not sure if this needs to explicitely specified in the method name to `throwIfUpdateValidationFails`. I mean isn't it implicit that the code should throw exception if it doesn't find something?

Reply:
no, it could return null and let the caller deal with the situation. Declarative naming of the agent methods helps the reader of the commandhandler to understand the execution flow even without looking into the agent  implementations.